src/projections/gnom.cpp

```diff
@@ -122,25 +124,105 @@ static PJ_LP gnom_s_inverse (PJ_XY xy, PJ *P) {           /* Spheroidal, inverse
     return lp;
 }
 
+static PJ_XY gnom_e_forward (PJ_LP lp, PJ *P) { /* Ellipsoidal, forward */
+    PJ_XY xy = {0.0,0.0};
+    struct pj_opaque *Q = static_cast<struct pj_opaque*>(P->opaque);
+
+    double
+        lat0 = P->phi0 / DEG_TO_RAD,
+        lon0 = 0,
+        lat1 = lp.phi / DEG_TO_RAD,
+        lon1 = lp.lam / DEG_TO_RAD,
+        azi0, m, M;
+
+    geod_geninverse(&Q->g, lat0, lon0, lat1, lon1,
+                    nullptr, &azi0, nullptr, &m, &M, nullptr, nullptr);
+    if (M <= 0) {
+        proj_errno_set(P, PROJ_ERR_COORD_TRANSFM_OUTSIDE_PROJECTION_DOMAIN);
+        xy.x = xy.y = HUGE_VAL;
+    }
+    else {
+        double rho = m/M;
+        azi0 *= DEG_TO_RAD;
+        xy.x = rho * sin(azi0);
+        xy.y = rho * cos(azi0);
```

Comment:
I've the feeling that geod_geninverse_int() returns those sin() and cos() values, so their re-computation could potentially be saved. Actually same pattern in aeqd

Reply:
this remark is more as a potential later optimization than something in the scope of this PR

Reply:
`geod_geninverse_int` is currently not exposed and I'm reluctant to do so to capture this particular optimization. At some future point (who knows when), the accuracy and speed of all PROJ code could probably be improved by routinely passing angles around as a `[sin/cos]` pair.  I started experimenting with this with the sample code for my paper on auxiliary latitudes.  See [the documentation on AuxAngle](https://geographiclib.sourceforge.io/C++/2.1.2/classGeographicLib_1_1AuxAngle.html).  (Ah, and I see your note about "potential future optimization".  So we are on the same page here.)